packages/teams-js/src/public/location.ts

```diff
@@ -48,25 +48,71 @@ export namespace location {
   }
 
   /**
-   * Fetches current user coordinates or allows user to choose location on map
-   *
-   * @param props {@link LocationProps} - Specifying how the location request is handled
-   * @returns Promise that will be fulfilled when the operation has completed
+   * Fetches current user coordinates
+   * @returns User's current location
    */
-  export function getLocation(props: LocationProps): Promise<Location>;
+  export function getCurrentLocation(): Promise<Location> {
+    ensureInitialized(FrameContexts.content, FrameContexts.task);
+    return sendAndHandleError('location.getLocation', { allowChooseLocation: false, showMap: false });
+  }
+
+  export function hasPermission(): Promise<boolean> {
+    return new Promise<boolean>(resolve => {
+      resolve(sendAndHandleError('location.hasPermission'));
+    });
+  }
+
+  // This should not trigger the "refresh the app scenario" because this is for setting things up
+  // for use through teamsjs-sdk 2.0. If the user DOES refresh the app after calling this the iframe
+  // would have the new allow parameters, but only the AppPermissions dialog should trigger the
+  // "ask the user to refresh" flow
+  export function requestPermission(): Promise<boolean> {
+    return new Promise<boolean>(resolve => {
+      resolve(sendAndHandleError('location.requestPermission'));
```

Comment:
would it be more scalable to use `requestPermission` as message type and send the kind of permissions being request as parameters? I'm trying to think about it form the perspective of host and host sdk. Because the flow when permissions is requested shouldn't differ (show UI, allow use to toggle a button) I would expect the host having a singular implementation for these. Which means having them all come across the isolation boundary as separate messages, with separate message handlers, is a bit unexpected.

Reply:
Same for `location.hasPermission` above.

Reply:
That makes sense to me, will tweak